Sources/Web3Wallet/Web3WalletClient.swift

```diff
@@ -190,19 +190,25 @@ public class Web3WalletClient {
     /// Query pending requests
     /// - Returns: Pending requests received from peer with `wc_sessionRequest` protocol method
     /// - Parameter topic: topic representing session for which you want to get pending requests. If nil, you will receive pending requests for all active sessions.
-    public func getPendingRequests(topic: String? = nil) -> [Request] {
+    public func getPendingRequests(topic: String? = nil) -> [(request: Request, context: VerifyContext?)] {
         signClient.getPendingRequests(topic: topic)
     }
     
+    /// Query pending proposals
+    /// - Returns: Pending proposals received from peer with `wc_sessionPropose` protocol method
+    public func getPendingProposals(topic: String? = nil) -> [(proposal: Session.Proposal, context: VerifyContext?)] {
+        signClient.getPendingProposals(topic: topic)
+    }
+    
     /// - Parameter id: id of a wc_sessionRequest jsonrpc request
     /// - Returns: json rpc record object for given id or nil if record for give id does not exits
-    public func getSessionRequestRecord(id: RPCID) -> Request? {
+    public func getSessionRequestRecord(id: RPCID) -> (request: Request, context: VerifyContext?)? {
```

Comment:
What do you think about separate `getVerifyContext(id: RPCID)`.
To not affect current Client interface